spec/es_spec_helper.rb

```diff
@@ -9,57 +9,61 @@
 CONTAINER_IMAGE = "elasticsearch"
 CONTAINER_TAG = "1.6"
 
-module ESHelper
+DOCKER_INTEGRATION = ENV["DOCKER_INTEGRATION"]
 
+module ESHelper
   def get_host
-    Longshoreman.new.get_host_ip
+    DOCKER_INTEGRATION ? Longshoreman.new.get_host_ip : "127.0.0.1"
   end
 
-  def get_port(protocol)
+  def get_port
+    return 9200 unless DOCKER_INTEGRATION
+
     container = Longshoreman::Container.new
     container.get(CONTAINER_NAME)
-    case protocol
-    when "http"
-      container.rport(9200)
-    when "transport", "node"
-      container.rport(9300)
-    end
+    container.rport(9200)
   end
 
   def get_client
-    Elasticsearch::Client.new(:host => "#{get_host}:#{get_port('http')}")
+    Elasticsearch::Client.new(:host => "#{get_host}:#{get_port}")
   end
 end
 
+
 RSpec.configure do |config|
   config.include ESHelper
 
-  # this :all hook gets run before every describe block that is tagged with :integration => true.
-  config.before(:all, :integration => true) do
-    # check if container exists already before creating new one.
-    begin
-      ls = Longshoreman::new
-      ls.container.get(CONTAINER_NAME)
-    rescue Docker::Error::NotFoundError
-      Longshoreman.new("#{CONTAINER_IMAGE}:#{CONTAINER_TAG}", CONTAINER_NAME)
-      # TODO(talevy): verify ES is running instead of static timeout
-      sleep 10
+
+  if DOCKER_INTEGRATION
+    # this :all hook gets run before every describe block that is tagged with :integration => true.
+    config.before(:all, :integration => true) do
+
+
+      # check if container exists already before creating new one.
+      begin
+        ls = Longshoreman::new
+        ls.container.get(CONTAINER_NAME)
+      rescue Docker::Error::NotFoundError
+        Longshoreman.new("#{CONTAINER_IMAGE}:#{CONTAINER_TAG}", CONTAINER_NAME)
+        # TODO(talevy): verify ES is running instead of static timeout
```

Comment:
making a note here for a possible follow-up PR.

maybe we can add an endpoint in the http_client wrapper we have going to check if ES is up. then we can loop-tick the check here until it says yes here as a replacement of the long-optimistic 10-second wait here